### ISSUE TITLE
A pod can't direct traffic to itself via a service

### ISSUE THREAD

**Issue by tvs** [cloudfoundry-incubator/kubo-release#212]
This [seems to be related to an issue with Flannel (and possible NSX-T) configuration](https://github.com/kubernetes/kubernetes/issues/20475). It might be as simple as enabling `hairpinMode` mode on Flannel.

```
$ kubectl run php-apache --image=k8s.gcr.io/hpa-example --requests=cpu=200m --expose --port=80
service "php-apache" created
deployment "php-apache" created

$ kubectl run -i --tty load-generator --image=busybox /bin/sh

Hit enter for command prompt

/ # wget http://php-apache.default.svc.cluster.local
Connecting to php-apache.default.svc.cluster.local (10.100.200.181:80)
index.html           100% |***********************************************************************************************|     3   0:00:00 ETA
/ # exit
Session ended, resume using 'kubectl attach load-generator-7bbbb4fdd4-c4tvn -c load-generator -i -t' command when the pod is running

$ kubectl exec -it php-apache-8699449574-ft848 -- /bin/sh
# curl http://php-apache.default.svc.cluster.local --verbose
* Rebuilt URL to: http://php-apache.default.svc.cluster.local/
* Hostname was NOT found in DNS cache
*   Trying 10.100.200.181...
# exit
```

**Comment by tvs** [comment]
This problem definitely affects NSX-T deployments as well. I also noticed that [there's a `hairpinMode` option for the Kubelet config](https://github.com/kubernetes/kubernetes/blob/release-1.10/pkg/kubelet/apis/kubeletconfig/v1beta1/types.go#L257-L268).

Hairpin traffic is one of the use-cases tested in [kubernetes/perf-tests/network](https://github.com/kubernetes/perf-tests/tree/master/network/benchmarks/netperf#implementation-and-test-methodology).

**Comment by alex-slynko** [comment]
Should be fixed as part of 0.21
